ansible_collections/arista/avd/roles/eos_designs/python_modules/overlay/router_bgp.py

```diff
@@ -440,6 +440,11 @@ def _create_neighbor(self, ip_address: str, name: str, peer_group: str, remote_a
 
             neighbor["remote_as"] = remote_as
 
+        if name in self._avd_overlay_peers:
+            peer_facts = self.shared_utils.get_peer_facts(name)
+            if self.shared_utils.shutdown_bgp_towards_undeployed_peers is True and peer_facts["is_deployed"] is False:
```

Comment:
Please move the first part of the if to be the very first thing we check above. So we don't have to `get_peer_facts` unneeded.

Reply:
Fixed